docs/technical_reference/unit_models/boron_removal.rst

```diff
@@ -0,0 +1,160 @@
+
+
+Chemical Dosing Parameters
+--------------------------
+In addition to providing and fixing values for the caustic additive, the users will
+need to provide a dictionary that maps specific species names from a property package
+to Boron, Borate, protons (optional), hydroxides (optional), and the cation from the
+caustic additive (optional). These are needed to establish the appropriate changes in
+the molar flow rate of each of these components and which variable they would apply to
+from the property package. Users may also optionally provide a specific name to the
+caustic chemical being added and are required to provide the molecular weight of the
+caustic additive, as well as the charge of the cation produced when the additive will
+dissociate (i.e., if the additive is NaOH, then the cation it dissociates into is
+Na\ :sup:`+`\ and it has a charge of 1).
+
+To provide this information to the unit model, users must add a 'chemical_mapping_data'
+dictionary to the initialization of the unit model. That dictionary must have the
+following format.
+
+.. code-block::
+
+   chem_dict = {
+                  'boron_name': 'name_of_species_representing_boron', #[is required]
+                  'borate_name': 'name_of_species_representing_borate', #[is required]
+                  'proton_name': 'name_of_species_representing_protons',  #[is optional]
+                  'hydroxide_name': 'name_of_species_representing_hydroxides', #[is optional]
+
+                  'caustic_additive':
+                  {
+                    'additive_name': 'name_of_the_actual_chemical', #[is optional]
+                    'cation_name': 'name_of_cation_species_in_additive', #[is optional]
+                    'mw_additive': (value, units), #[is required]
+                    'charge_additive': value, #[is required]
+                  }
+               }
+
+For example, this 'chem_dict' would be passed into the model on construction as
+one of the configuration options as shown below.
+
+.. code-block::
+
+    model.fs.unit = BoronRemoval(
+            default={
+                "property_package": model.fs.properties,
+                "chemical_mapping_data": chem_dict,
+            }
+        )
+
+**NOTE: The above example assumes you have already constructed a pyomo model named 'model' and attached an IDAES flowsheet named 'fs' to it, as well as a properties block named 'properties'**
+
+Equations and Relationships
+---------------------------
+Reaction constants :math:`K_{w}` and :math:`K_{a}` are taken from the Benjamin (2010) chemistry
+textbook (see References below).
+
+.. csv-table::
+   :header: "Description", "Equation"
+
+   "Electroneutrality Constraint", ":math:`C_{H} = C_{OH} + C_{A} + Alk_{Res} - n_{caustic}*C_{caustic}`"
+   "Residual Alkalinity Expression", ":math:`Alk_{Res} = -{\sum_{i} n_i \cdotp C_{out,i} }`"

```

Comment:
Is index `i` equivalent to `j` ?  This might have caused my confusion about the neutrality and alkalinity equations in the table.  Are `i` only those buffering species (to make sense of the alkalinity equation) ?  This should probably be clarified better.  Maybe also add clarification for the exact alkalinity definition here, i.e., total or carbonate.

Reply:
Added a note below the table.

Reply:
Related to my comment in the next lines, I think the note of "stream after the first RO" isn't agreeing with this equation too, which calculates on the outlet of this pH increasing unit.  Did I miss something?

Reply:
Related to my comment in the next lines, I think the note of "stream after the first RO" isn't agreeing with this equation too, which calculates on the outlet of this pH increasing unit.  Did I miss something?